subprojects/snapshots/src/main/java/org/gradle/internal/fingerprint/FileSystemLocationFingerprint.java

```diff
@@ -31,7 +31,6 @@
  */
 public interface FileSystemLocationFingerprint extends Comparable<FileSystemLocationFingerprint>, Hashable {
     HashCode DIR_SIGNATURE = Hashing.signature("DIR");
-    HashCode MISSING_FILE_SIGNATURE = Hashing.signature("MISSING");
```

Comment:
Why move this? (And then why not move `DIR_SIGNATURE` too?)

Looks like we are mixing fingerprints and snapshots here.

Reply:
We need a single instance of each, they're comapred with `==`.
Didn't find another place for `DIR_SIGNATURE` being defined.

Reply:
Is it okay that the `HashCode` value of `MISSING_FILE_SIGNATURE` has changed now?
I understood that it's only important to be consistent during a run, is that right?

Reply:
So this is a signature for a _fingerprint,_ while the one you replaced it with was a signature for a _snapshot._ I think we shouldn't mix them, as they are separate concepts. Let's keep the constants separate, too.

Reply:
Thanks, reverted